# FAD-Multiomics-Manuscript

[![HTML Manuscript](https://img.shields.io/badge/manuscript-HTML-blue.svg)](https://doi.org/10.1186/s13195-024-01659-6)
[![PDF Manuscript](https://img.shields.io/badge/manuscript-PDF-blue.svg)](https://alzres.biomedcentral.com/counter/pdf/10.1186/s13195-024-01659-6.pdf)

If you use the code in this repository for a publication, please cite the following publication:

"Phoebe Valdes*, Andrew B. Caldwell*, Qing Liu, Michael Q. Fitzgerald, Srinivasan Ramachandran, Celeste M. Karch, the Dominantly Inherited Alzheimer Network (DIAN), Douglas R. Galasko, Shauna H. Yuan, Steven L. Wagner, and Shankar Subramaniam. Integrative multiomics reveals common endotypes across PSEN1, PSEN2, and APP mutations in familial Alzheimer’s disease. Alz Res Therapy 17, 5 (2025)."

## Manuscript Abstract
<i>PSEN1</i>, <i>PSEN2</i>, and <i>APP</i> mutations cause Alzheimer’s disease (AD) with an early age at onset (AAO) and progressive cognitive decline. Although <i>PSEN1</i> mutations are more com-mon and generally have an earlier AAO, certain mutations in <i>PSEN1</i> cause a later AAO, similar to those in <i>PSEN2</i> and <i>APP</i>. In this work, we examined whether common disease endo-types exist across these mutations with a later AAO (average age of ~55 years) using hiPSC-derived neurons from autosomal-dominant, familial Alzheimer’s disease (FAD) patients harboring mutations in <i>PSEN1<sup>A79V</sup></i>, <i>PSEN2<sup>N141I</sup></i>, and <i>APP<sup>V717I</sup></i>. Using RNA-seq and ATAC-seq, we mechanistically characterized these FAD mutations in patient-derived neurons and identified the correlation between gene expression and chromatin accessibility associated with key disease endotypes. Furthermore, we identified common endotypes across the three FAD mutations: dedifferentiation of a mature neuron to a less differentiated quasi-neuron state, dysregulation of synaptic signaling, repression of mitochondrial function and metabolism, and inflammation. Integrative analysis allowed us to ascertain the master transcriptional regulators associated with these endotypes, including REST, ASCL1, and ZIC family members (activation), as well as NRF1 (repression). Our findings demonstrate that FAD mutations share common regulatory changes within endotypes with varying severity, leading to the reversion to a less-differentiated neuron state. The regulatory mechanisms described here offer potential targets for therapeutic interventions.

## Repository Description
Sequencing analysis scripts for Valdes et. al 2025 FAD Multiomics Manuscript.

## Data Availability
The sequencing data for this project is available at [![HTML GEO](https://img.shields.io/badge/GEO-GSE206603-blue)](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE206603)
